internal/profiles/service.go

```diff
@@ -185,6 +182,9 @@ func (p *profileService) CreateProfile(
 	profile.Remediate = ptr.Ptr(string(newProfile.Remediate.ActionType))
 	profile.Alert = ptr.Ptr(string(newProfile.Alert.ActionType))
 
+	// Adds default rule names, if not present
+	PopulateRuleNames(profile, rulesInProf)
+
```

Comment:
Moving this down here means that we're now storing a `NULL` for the rule name if it's not set.  Do we also need to call `PopulateRuleNames` on the Get / List output?  (Alternatively, moving this above line 155 should fix this problem.)

Reply:
It looks like we didn't move profile listing into the `profiles` service, and we just call some methods like `MergeDatabaseListIntoProfiles` directly from `handlers_profile.go` without any support for mocking.  (And yes, that means the code there isn't tested...)

Reply:
It's intentionally moved to the end of this function to be near the other calls that populate the result.

> Moving this down here means that we're now storing a NULL for the rule name if it's not set

The display name will always be set, because it's defaulted to the rule type name.

> Do we also need to call PopulateRuleNames on the Get / List output?

No, the correct name is already stored in the DB during the create / update and fetched during the get / list.